### PR TITLE
[EXP] generalise the use of fstrings (py36+)

### PR DESCRIPTION
**Pull request by neutrinoceros**
## PR Summary

apply [flynt](https://github.com/ikamensh/flynt) to automatically use strings everywhere.

### CODE REVIEW

File: doc/extensions/yt_cookbook.py [hunk 1]
```diff
@@ -49,29 +49,29 @@ def run(self):
...
-        lines.append(".. literalinclude:: %s" % self.arguments[0])
+        lines.append(f".. literalinclude:: {self.arguments[0]}")
         lines.append("\n")
         lines.append("\n")
         for im in images:
             im_name = os.path.join("_static", os.path.basename(im))
-            lines.append(".. image:: %s" % im_name)
+            lines.append(f".. image:: {im_name}")
             lines.append("   :width: 400")
             lines.append("   :target: ../_images/%s" % os.path.basename(im))
```
[the remark this answers is not in the document]
Comment (neutrinoceros) [reply doc/extensions/yt_cookbook.py:64]: Ah, I just spotted that those are doc files, which is probably why they are never run on windows.

[the remark this answers is not in the document]
Comment (neutrinoceros) [reply doc/extensions/yt_cookbook.py:64]: Not the bug I suspected, but a bug nonetheless, I reported it to Flynt's author.

[the remark this answers is not in the document]
Comment (neutrinoceros) [reply doc/extensions/yt_cookbook.py]: the bug is fixed upstream (flynt v0.47.1) !

**Comment by neutrinoceros** [comment]
Just a quick note that `logging` is designed to postpone `%` formatting until entries are needed (according to log level), so switching to strings here causes a performance hit of variable importance (depending on the evaluation cost of variables that end up in the log entry).
This seems like a non-trivial feature to add to flynt, so I don't think it's realistic to wait on it :/

**Comment by neutrinoceros** [comment]
not going to go forward with this for now

**Comment by matthewturk** [comment]
Ah, that's a shame, I was super excited about this.

**Comment by neutrinoceros** [comment]
Well the problem is that I don't know how to special case the logger messages as flynt doesn't discriminate them, so for now it looks like much more manual work than I was hoping for originally. Plus I had no idea anyone but me was interested in this until now :)

**Comment by munkm** [comment]
I think it would be worth opening an issue for this so we keep it in the queue!

**Pull request closed without merging**